scheduling/job_scheduling.py

```diff
@@ -1,112 +1,85 @@
+"""
 Author : Mohit Kumar
 Job Sequencing Problem implemented in python
-'''
+"""
 
 
 class Scheduling:
-
     def __init__(self, jobs):
         self.jobs = jobs
 
     def schedule(self, n, deadline):
         self.sdl = deadline
-        self.J = []
-        self.J.append(self.jobs[1])
+        self.J = [self.jobs[1]]
         self.x = 2
         while self.x < n:
             self.K = self.J.copy()
             self.K.append(self.jobs[self.x])
             self.x = self.x + 1
-            self.feasibility = self.feasible(self.K, self.sdl)
-            if self.feasibility is True:
+            if self.feasible(self.K, self.sdl):
                 self.J = self.K.copy()
 
         return self.J
 
     def feasible(self, K, fdl):
         self.tmp = K
-        self.isFeasible = True
+        self.is_feasible = True
 
-        self.i = 0
-        self.j = 1
-        self.k = 0
+        i = 0
+        j = 1
+        k = 0
 
-        while self.i < len(self.tmp):
-            while self.j < len(self.tmp):
-                self.index1 = self.jobs.index(self.tmp[self.i])
-                self.index2 = self.jobs.index(self.tmp[self.j])
-                self.j = self.j + 1
+        while i < len(self.tmp):
+            while j < len(self.tmp):
+                self.index1 = self.jobs.index(self.tmp[i])
+                self.index2 = self.jobs.index(self.tmp[j])
+                j += 1
                 if fdl[self.index1] > fdl[self.index2]:
-                    (self.tmp[self.i], self.tmp[self.j]) = \
-                        (self.tmp[self.j], self.tmp[self.i])
-            self.i = self.i + 1
-
-        while self.k < len(self.tmp):
-            self.job = self.tmp[self.k]
+                    (self.tmp[i], self.tmp[j]) = (
+                        self.tmp[j],
+                        self.tmp[i],
+                    )
+            i += 1
+
+        while k < len(self.tmp):
+            self.job = self.tmp[k]
             self.jobindex = self.jobs.index(self.job)
             self.dlineval = fdl[self.jobindex]
-            self.ftest = self.k + 1
-            self.k = self.k + 1
+            self.ftest = k + 1
+            k += 1
             if self.dlineval < self.ftest:
-                self.isFeasible = False
+                self.is_feasible = False
                 break
 
-        return self.isFeasible
-
+        return self.is_feasible
 
 def main():
     n = 7  # Number of jobs
-    jobs = [  # jobs id
-        0,
-        1,
-        2,
-        3,
-        4,
-        5,
-        6
-    ]
-    deadline = [  # deadline to job[i]
-        0,
-        2,
-        4,
-        3,
-        3,
-        2,
-        1
-    ]
-    profit = [  # profit associated wrt job[i]
-        0,
-        46,
-        52,
-        30,
-        36,
-        56,
-        40
-    ]
+    jobs = [0, 1, 2, 3, 4, 5, 6]  # jobs id
+    deadline = [0, 2, 4, 3, 3, 2, 1]  # deadline to job[i]
+    profit = [0, 46, 52, 30, 36, 56, 40]  # profit associated wrt job[i]
 
```

Comment:
Organize job data using a namedtuple...
```python
from collections import namedtuple
job = namedtuple("job", "job_id deadline profit")
jobs = [job(0, 0, 0), job(1, 2, 46) ... ]
```

Reply:
Actually I am not familiar with namedtuple and how to use it so i need  a bit more time to apply that